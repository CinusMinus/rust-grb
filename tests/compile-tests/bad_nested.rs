use grb::*;
mod utils;

fn main() -> Result<()> {
  create_model!(_g, m, x, y, z);
  c!(x + y == 2 == 1 - z);
  c!(x + y - (4 >= 1 - z));
  c!(x + y >= 1 - z >= 43);
  Ok(())
}
