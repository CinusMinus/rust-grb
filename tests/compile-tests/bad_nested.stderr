error: expression should be a ==, >= or <= comparison
 --> $DIR/bad_nested.rs:7:6
  |
7 |   c!(x + y - (4 >= 1 - z));
  |      ^^^^^^^^^^^^^^^^^^^^

error[E0369]: binary operation `==` cannot be applied to type `Expr`
 --> $DIR/bad_nested.rs:6:12
  |
6 |   c!(x + y == 2 == 1 - z);
  |      ----- ^^ - {integer}
  |      |
  |      Expr

error[E0369]: binary operation `>=` cannot be applied to type `Expr`
 --> $DIR/bad_nested.rs:8:12
  |
8 |   c!(x + y >= 1 - z >= 43);
  |      ----- ^^ ----- _
  |      |
  |      Expr
